### PR TITLE
[DOCFIX] Fix doc for Flink

### PR DESCRIPTION
**Pull request by riversand963**

### CODE REVIEW

**Comment by riversand963** [comment]
@aaudiber 
Could you take a look?
Also do you think it is necessary to explain how to "Translate additional Alluxio site properties to Flink", which is mentioned in the doc?

File: docs/en/Running-Flink-on-Alluxio.md [hunk 1]
```diff
@@ -67,7 +68,7 @@ available on all cluster nodes as well). For example like this:
 
 {% include Running-Flink-on-Alluxio/hadoop-classpath.md %}
 
-Alternatively, advanced users can choose to compile this client jar from the source code. Follow the instructs [here](Building-Alluxio-Master-Branch.html#compute-framework-support) and use the generated jar at `{{site.ALLUXIO_CLIENT_JAR_PATH_BUILD}}` for the rest of this guide.
+Alternatively, advanced users can choose to compile this client jar from the source code. Follow the instructs [here](Building-Alluxio-Master-Branch.html#compute-framework-support) and use the generated jar at `{{site.ALLUXIO_CLIENT_JAR_PATH}}` for the rest of this guide.
```
Comment (aaudiber) [inline docs/en/Running-Flink-on-Alluxio.md]: instructs -> instructions

File: docs/en/Running-Flink-on-Alluxio.md [hunk 2]
```diff
@@ -48,8 +48,9 @@ the directory usually ends with `etc/hadoop`.)
 ### Generate and Distribute the Alluxio Client Jar
 
 In order to communicate with Alluxio, we need to provide Flink programs with the Alluxio Core Client
-jar.
-Generate the Flink compatible client jar by building the entire project from the top level `alluxio`
+jar. If you are using Alluxio binary, then compilation is unnecessary and the jar is available at
```
Comment (aaudiber) [inline docs/en/Running-Flink-on-Alluxio.md]: They shouldn't need to rebuild the jar - we generate a flink client at ${ALLUXIO_HOME}/client/flink/alluxio-${{ALLUXIO_RELEASED_VERSION}}-flink-client.jar

**Comment by aaudiber** [comment]
@riversand963  +1 for giving a better explanation for how to translate Alluxio properties into Flink

File: docs/en/Running-Flink-on-Alluxio.md (hunk 1 above)
Comment (riversand963) [reply docs/en/Running-Flink-on-Alluxio.md]: Done.

**Comment by riversand963** [comment]
@aaudiber It's not clear what alluxio-site properties need to go into flink-conf.yaml. The doc says all site-properties should, but according to my test, even if we do not specify the properties in flink-conf.yaml, the example can still run.

File: docs/en/Running-Flink-on-Alluxio.md (hunk 2 above)
Comment (aaudiber) [reply docs/en/Running-Flink-on-Alluxio.md]: The primary audience for docs is the people using a distribution. I think the footnote "Alternatively, advanced users can choose ..." below is good enough to help those who want to build from source

**Comment by aaudiber** [comment]
There aren't any required properties, but if the user wants to configure the behavior of the Alluxio client they need a way to supply properties. For example, they might want to use CACHE_THROUGH as the write type, so they want to set `alluxio.user.file.writetype.default=CACHE_THROUGH`, or if they are using Alluxio in HA-mode, they need to specify zookeeper configuration like `alluxio.zookeeper.address`, etc.

File: docs/en/Building-Alluxio-Master-Branch.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ To have the logs output to STDOUT, append the following to the `mvn` command
 {% include Building-Alluxio-Master-Branch/STDOUT.md %}
 
 ## Compute Framework Support
-To build Alluxio against different compute frameworks, you can run Maven build with different compute profiles. The generated Alluxio client is located at `{{site.ALLUXIO_CLIENT_JAR_PATH_BUILD}}`.
+To build Alluxio against different compute frameworks, you can run Maven build with different compute profiles. The generated Alluxio client is located at `{{site.ALLUXIO_CLIENT_JAR_PATH}}`.
```
Comment (aaudiber) [inline docs/en/Building-Alluxio-Master-Branch.md]: This actually was supposed to be `ALLUXIO_CLIENT_JAR_PATH_BUILD` - it will be fixed by https://github.com/Alluxio/alluxio/pull/5990/files so you can undo the change here

File: docs/en/Running-Flink-on-Alluxio.md [hunk 4]
```diff
@@ -48,8 +48,9 @@ the directory usually ends with `etc/hadoop`.)
...
+`{{site.ALLUXIO_CLIENT_JAR_PATH}}`. If you want to build from Alluxio source code, then you need to
```
Comment (aaudiber) [inline docs/en/Running-Flink-on-Alluxio.md]: `site.ALLUXIO_CLIENT_JAR_PATH_FLINK`

File: docs/_includes/Running-Flink-on-Alluxio/alluxio-client-properties.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+```yaml
+env.java.opts: -Dalluxio.user.file.writetype.default=CACHE_THROUGH
```
Comment (aaudiber) [inline docs/_includes/Running-Flink-on-Alluxio/alluxio-client-properties.md]: we're trying to move away from creating external code snippets - it makes reading and editing the docs source code much harder. Please inline this snippet

File: docs/en/Running-Flink-on-Alluxio.md [hunk 6]
```diff
@@ -67,13 +68,18 @@ available on all cluster nodes as well). For example like this:
...
+[here](Building-Alluxio-Master-Branch.html#compute-framework-support) and use the generated jar at
+`{{site.ALLUXIO_CLIENT_JAR_PATH}}` for the rest of this guide.
 
 ### Translate additional Alluxio site properties to Flink
 
-In addition, if there are any properties specified in `conf/alluxio-site.properties`,
+In addition, if there are any client-related properties specified in `conf/alluxio-site.properties`,
 translate those to `env.java.opts` in `{FLINK_HOME}/conf/flink-conf.yaml` for Flink to pick up
-Alluxio configuration.
+Alluxio configuration. For example, if you want to configure Alluxio client to use CACHE\_THROUGH as
```
Comment (aaudiber) [inline docs/en/Running-Flink-on-Alluxio.md]: `CACHE_THROUGH`

File: docs/en/Running-Flink-on-Alluxio.md [hunk 7]
```diff
@@ -67,13 +68,18 @@ available on all cluster nodes as well). For example like this:
 
 {% include Running-Flink-on-Alluxio/hadoop-classpath.md %}
 
-Alternatively, advanced users can choose to compile this client jar from the source code. Follow the instructs [here](Building-Alluxio-Master-Branch.html#compute-framework-support) and use the generated jar at `{{site.ALLUXIO_CLIENT_JAR_PATH_BUILD}}` for the rest of this guide.
+Alternatively, advanced users can choose to compile this client jar from the source code. Follow the instructions
+[here](Building-Alluxio-Master-Branch.html#compute-framework-support) and use the generated jar at
+`{{site.ALLUXIO_CLIENT_JAR_PATH}}` for the rest of this guide.
```
Comment (apc999) [inline docs/en/Running-Flink-on-Alluxio.md]: `{{site.ALLUXIO_CLIENT_JAR_PATH_BUILD}}`

Comment (riversand963) [reply docs/en/Running-Flink-on-Alluxio.md]: Done.

File: docs/en/Running-Flink-on-Alluxio.md (hunk 6 above)
Comment (riversand963) [reply docs/en/Running-Flink-on-Alluxio.md]: Done.

File: docs/_includes/Running-Flink-on-Alluxio/alluxio-client-properties.md (hunk 5 above)
Comment (riversand963) [reply docs/_includes/Running-Flink-on-Alluxio/alluxio-client-properties.md]: Done.

File: docs/en/Building-Alluxio-Master-Branch.md (hunk 3 above)
Comment (riversand963) [reply docs/en/Building-Alluxio-Master-Branch.md]: Done.

File: docs/en/Running-Flink-on-Alluxio.md (hunk 2 above)
Comment (riversand963) [reply docs/en/Running-Flink-on-Alluxio.md]: Done.

File: docs/en/Running-Flink-on-Alluxio.md [hunk 8]
```diff
@@ -45,15 +45,11 @@ Next, you have to specify the path to the Hadoop configuration in Flink. Open th
 configuration value to the **directory** containing the `core-site.xml`. (For newer Hadoop versions,
 the directory usually ends with `etc/hadoop`.)
 
-### Generate and Distribute the Alluxio Client Jar
+### Distribute the Alluxio Client Jar
 
 In order to communicate with Alluxio, we need to provide Flink programs with the Alluxio Core Client
 jar. If you are using Alluxio binary, then compilation is unnecessary and the jar is available at
```
Comment (aaudiber) [inline docs/en/Running-Flink-on-Alluxio.md]: We can leave out "compilation is unnecessary" - Users of the distribution tarball wouldn't expect to need to compile

Comment (riversand963) [reply docs/en/Running-Flink-on-Alluxio.md]: Done.
